[2021-09-17 18:00:43,661] INFO [GroupMetadataManager brokerId=0] Removed 0 expired offsets in 1 milliseconds. (kafka.coordinator.group.GroupMetadataManager)
[2021-09-17 18:00:52,866] INFO [GroupCoordinator 0]: Member[group.instance.id None, member.id consumer-3-7b7c6c71-881b-40a7-8219-ce94ec31fb08] in group product has left, removing it from the group (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:00:52,866] INFO [GroupCoordinator 0]: Preparing to rebalance group product in state PreparingRebalance with old generation 3 (__consumer_offsets-15) (reason: removing member consumer-3-7b7c6c71-881b-40a7-8219-ce94ec31fb08 on LeaveGroup) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:00:52,867] INFO [GroupCoordinator 0]: Group product with generation 4 is now empty (__consumer_offsets-15) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:12,350] INFO [GroupCoordinator 0]: Dynamic Member with unknown member id joins group product in Empty state. Created a new member id consumer-3-fb1e339b-5ab7-4213-a6a3-c91a7a9bbaf4 for this member and add to the group. (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:12,350] INFO [GroupCoordinator 0]: Preparing to rebalance group product in state PreparingRebalance with old generation 4 (__consumer_offsets-15) (reason: Adding new member consumer-3-fb1e339b-5ab7-4213-a6a3-c91a7a9bbaf4 with group instance id None) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:12,351] INFO [GroupCoordinator 0]: Stabilized group product generation 5 (__consumer_offsets-15) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:12,355] INFO [GroupCoordinator 0]: Assignment received from leader for group product for generation 5 (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:24,485] INFO [GroupCoordinator 0]: Member[group.instance.id None, member.id consumer-3-0499a24f-66a4-4de4-9045-7dfc43994dba] in group delivery has left, removing it from the group (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:24,486] INFO [GroupCoordinator 0]: Preparing to rebalance group delivery in state PreparingRebalance with old generation 5 (__consumer_offsets-46) (reason: removing member consumer-3-0499a24f-66a4-4de4-9045-7dfc43994dba on LeaveGroup) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:24,486] INFO [GroupCoordinator 0]: Group delivery with generation 6 is now empty (__consumer_offsets-46) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:35,164] INFO [GroupCoordinator 0]: Member[group.instance.id None, member.id consumer-3-95890c02-2141-4e02-8a38-f65f3b457800] in group order has left, removing it from the group (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:35,164] INFO [GroupCoordinator 0]: Preparing to rebalance group order in state PreparingRebalance with old generation 11 (__consumer_offsets-0) (reason: removing member consumer-3-95890c02-2141-4e02-8a38-f65f3b457800 on LeaveGroup) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:35,164] INFO [GroupCoordinator 0]: Group order with generation 12 is now empty (__consumer_offsets-0) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:37,295] INFO [GroupCoordinator 0]: Dynamic Member with unknown member id joins group delivery in Empty state. Created a new member id consumer-3-66d56f65-f32a-4e3b-9ede-a873bd24274d for this member and add to the group. (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:37,295] INFO [GroupCoordinator 0]: Preparing to rebalance group delivery in state PreparingRebalance with old generation 6 (__consumer_offsets-46) (reason: Adding new member consumer-3-66d56f65-f32a-4e3b-9ede-a873bd24274d with group instance id None) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:37,297] INFO [GroupCoordinator 0]: Stabilized group delivery generation 7 (__consumer_offsets-46) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:37,301] INFO [GroupCoordinator 0]: Assignment received from leader for group delivery for generation 7 (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:47,595] INFO [GroupCoordinator 0]: Dynamic Member with unknown member id joins group order in Empty state. Created a new member id consumer-3-2465d555-bc81-4db7-a620-6abec17830d9 for this member and add to the group. (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:47,596] INFO [GroupCoordinator 0]: Preparing to rebalance group order in state PreparingRebalance with old generation 12 (__consumer_offsets-0) (reason: Adding new member consumer-3-2465d555-bc81-4db7-a620-6abec17830d9 with group instance id None) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:47,597] INFO [GroupCoordinator 0]: Stabilized group order generation 13 (__consumer_offsets-0) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:47,600] INFO [GroupCoordinator 0]: Assignment received from leader for group order for generation 13 (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:01:50,937] ERROR Closing socket for 127.0.0.1:9092-127.0.0.1:33894-55 because of error (kafka.network.Processor)
org.apache.kafka.common.errors.InvalidRequestException: Error parsing request header. Our best guess of the apiKey is: 0
Caused by: java.nio.BufferUnderflowException
	at java.base/java.nio.Buffer.nextGetIndex(Buffer.java:651)
	at java.base/java.nio.HeapByteBuffer.getShort(HeapByteBuffer.java:350)
	at org.apache.kafka.common.protocol.ByteBufferAccessor.readShort(ByteBufferAccessor.java:38)
	at org.apache.kafka.common.message.RequestHeaderData.read(RequestHeaderData.java:119)
	at org.apache.kafka.common.message.RequestHeaderData.<init>(RequestHeaderData.java:83)
	at org.apache.kafka.common.requests.RequestHeader.parse(RequestHeader.java:93)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1(SocketServer.scala:875)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1$adapted(SocketServer.scala:871)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at kafka.network.Processor.processCompletedReceives(SocketServer.scala:871)
	at kafka.network.Processor.run(SocketServer.scala:761)
	at java.base/java.lang.Thread.run(Thread.java:829)
[2021-09-17 18:01:50,941] ERROR Exception while processing request from 127.0.0.1:9092-127.0.0.1:33894-55 (kafka.network.Processor)
org.apache.kafka.common.errors.InvalidRequestException: Error parsing request header. Our best guess of the apiKey is: 0
Caused by: java.nio.BufferUnderflowException
	at java.base/java.nio.Buffer.nextGetIndex(Buffer.java:651)
	at java.base/java.nio.HeapByteBuffer.getShort(HeapByteBuffer.java:350)
	at org.apache.kafka.common.protocol.ByteBufferAccessor.readShort(ByteBufferAccessor.java:38)
	at org.apache.kafka.common.message.RequestHeaderData.read(RequestHeaderData.java:119)
	at org.apache.kafka.common.message.RequestHeaderData.<init>(RequestHeaderData.java:83)
	at org.apache.kafka.common.requests.RequestHeader.parse(RequestHeader.java:93)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1(SocketServer.scala:875)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1$adapted(SocketServer.scala:871)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at kafka.network.Processor.processCompletedReceives(SocketServer.scala:871)
	at kafka.network.Processor.run(SocketServer.scala:761)
	at java.base/java.lang.Thread.run(Thread.java:829)
[2021-09-17 18:02:03,002] ERROR Closing socket for 127.0.0.1:9092-127.0.0.1:33896-55 because of error (kafka.network.Processor)
org.apache.kafka.common.errors.InvalidRequestException: Error parsing request header. Our best guess of the apiKey is: 0
Caused by: java.nio.BufferUnderflowException
	at java.base/java.nio.Buffer.nextGetIndex(Buffer.java:651)
	at java.base/java.nio.HeapByteBuffer.getShort(HeapByteBuffer.java:350)
	at org.apache.kafka.common.protocol.ByteBufferAccessor.readShort(ByteBufferAccessor.java:38)
	at org.apache.kafka.common.message.RequestHeaderData.read(RequestHeaderData.java:119)
	at org.apache.kafka.common.message.RequestHeaderData.<init>(RequestHeaderData.java:83)
	at org.apache.kafka.common.requests.RequestHeader.parse(RequestHeader.java:93)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1(SocketServer.scala:875)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1$adapted(SocketServer.scala:871)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at kafka.network.Processor.processCompletedReceives(SocketServer.scala:871)
	at kafka.network.Processor.run(SocketServer.scala:761)
	at java.base/java.lang.Thread.run(Thread.java:829)
[2021-09-17 18:02:03,003] ERROR Exception while processing request from 127.0.0.1:9092-127.0.0.1:33896-55 (kafka.network.Processor)
org.apache.kafka.common.errors.InvalidRequestException: Error parsing request header. Our best guess of the apiKey is: 0
Caused by: java.nio.BufferUnderflowException
	at java.base/java.nio.Buffer.nextGetIndex(Buffer.java:651)
	at java.base/java.nio.HeapByteBuffer.getShort(HeapByteBuffer.java:350)
	at org.apache.kafka.common.protocol.ByteBufferAccessor.readShort(ByteBufferAccessor.java:38)
	at org.apache.kafka.common.message.RequestHeaderData.read(RequestHeaderData.java:119)
	at org.apache.kafka.common.message.RequestHeaderData.<init>(RequestHeaderData.java:83)
	at org.apache.kafka.common.requests.RequestHeader.parse(RequestHeader.java:93)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1(SocketServer.scala:875)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1$adapted(SocketServer.scala:871)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at kafka.network.Processor.processCompletedReceives(SocketServer.scala:871)
	at kafka.network.Processor.run(SocketServer.scala:761)
	at java.base/java.lang.Thread.run(Thread.java:829)
[2021-09-17 18:02:16,009] ERROR Closing socket for 127.0.0.1:9092-127.0.0.1:33898-55 because of error (kafka.network.Processor)
org.apache.kafka.common.errors.InvalidRequestException: Error parsing request header. Our best guess of the apiKey is: 0
Caused by: java.nio.BufferUnderflowException
	at java.base/java.nio.Buffer.nextGetIndex(Buffer.java:651)
	at java.base/java.nio.HeapByteBuffer.getShort(HeapByteBuffer.java:350)
	at org.apache.kafka.common.protocol.ByteBufferAccessor.readShort(ByteBufferAccessor.java:38)
	at org.apache.kafka.common.message.RequestHeaderData.read(RequestHeaderData.java:119)
	at org.apache.kafka.common.message.RequestHeaderData.<init>(RequestHeaderData.java:83)
	at org.apache.kafka.common.requests.RequestHeader.parse(RequestHeader.java:93)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1(SocketServer.scala:875)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1$adapted(SocketServer.scala:871)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at kafka.network.Processor.processCompletedReceives(SocketServer.scala:871)
	at kafka.network.Processor.run(SocketServer.scala:761)
	at java.base/java.lang.Thread.run(Thread.java:829)
[2021-09-17 18:02:16,010] ERROR Exception while processing request from 127.0.0.1:9092-127.0.0.1:33898-55 (kafka.network.Processor)
org.apache.kafka.common.errors.InvalidRequestException: Error parsing request header. Our best guess of the apiKey is: 0
Caused by: java.nio.BufferUnderflowException
	at java.base/java.nio.Buffer.nextGetIndex(Buffer.java:651)
	at java.base/java.nio.HeapByteBuffer.getShort(HeapByteBuffer.java:350)
	at org.apache.kafka.common.protocol.ByteBufferAccessor.readShort(ByteBufferAccessor.java:38)
	at org.apache.kafka.common.message.RequestHeaderData.read(RequestHeaderData.java:119)
	at org.apache.kafka.common.message.RequestHeaderData.<init>(RequestHeaderData.java:83)
	at org.apache.kafka.common.requests.RequestHeader.parse(RequestHeader.java:93)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1(SocketServer.scala:875)
	at kafka.network.Processor.$anonfun$processCompletedReceives$1$adapted(SocketServer.scala:871)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:553)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:551)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:921)
	at kafka.network.Processor.processCompletedReceives(SocketServer.scala:871)
	at kafka.network.Processor.run(SocketServer.scala:761)
	at java.base/java.lang.Thread.run(Thread.java:829)
[2021-09-17 18:02:39,499] INFO [GroupCoordinator 0]: Member[group.instance.id None, member.id consumer-3-66d56f65-f32a-4e3b-9ede-a873bd24274d] in group delivery has left, removing it from the group (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:02:39,499] INFO [GroupCoordinator 0]: Preparing to rebalance group delivery in state PreparingRebalance with old generation 7 (__consumer_offsets-46) (reason: removing member consumer-3-66d56f65-f32a-4e3b-9ede-a873bd24274d on LeaveGroup) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:02:39,499] INFO [GroupCoordinator 0]: Group delivery with generation 8 is now empty (__consumer_offsets-46) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:02:51,381] INFO [GroupCoordinator 0]: Dynamic Member with unknown member id joins group delivery in Empty state. Created a new member id consumer-3-f32f6425-54e1-4e6d-9c77-fbe33a08cc05 for this member and add to the group. (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:02:51,381] INFO [GroupCoordinator 0]: Preparing to rebalance group delivery in state PreparingRebalance with old generation 8 (__consumer_offsets-46) (reason: Adding new member consumer-3-f32f6425-54e1-4e6d-9c77-fbe33a08cc05 with group instance id None) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:02:51,381] INFO [GroupCoordinator 0]: Stabilized group delivery generation 9 (__consumer_offsets-46) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:02:51,384] INFO [GroupCoordinator 0]: Assignment received from leader for group delivery for generation 9 (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:03:45,555] INFO [GroupCoordinator 0]: Member[group.instance.id None, member.id consumer-3-2465d555-bc81-4db7-a620-6abec17830d9] in group order has left, removing it from the group (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:03:45,555] INFO [GroupCoordinator 0]: Preparing to rebalance group order in state PreparingRebalance with old generation 13 (__consumer_offsets-0) (reason: removing member consumer-3-2465d555-bc81-4db7-a620-6abec17830d9 on LeaveGroup) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:03:45,555] INFO [GroupCoordinator 0]: Group order with generation 14 is now empty (__consumer_offsets-0) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:03:57,101] INFO [GroupCoordinator 0]: Dynamic Member with unknown member id joins group order in Empty state. Created a new member id consumer-3-a9410ed4-c359-41db-a2b0-b624bc5d139a for this member and add to the group. (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:03:57,101] INFO [GroupCoordinator 0]: Preparing to rebalance group order in state PreparingRebalance with old generation 14 (__consumer_offsets-0) (reason: Adding new member consumer-3-a9410ed4-c359-41db-a2b0-b624bc5d139a with group instance id None) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:03:57,101] INFO [GroupCoordinator 0]: Stabilized group order generation 15 (__consumer_offsets-0) (kafka.coordinator.group.GroupCoordinator)
[2021-09-17 18:03:57,105] INFO [GroupCoordinator 0]: Assignment received from leader for group order for generation 15 (kafka.coordinator.group.GroupCoordinator)
